lib/FieldType/TranslatableLabel.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\RepositoryForms\FieldType;
+
+trait TranslatableLabel
+{
+    private function resolveLabel(array $names, array $languageCodes): string
+    {
+        foreach ($languageCodes as $languageCode) {
+            if (array_key_exists($languageCode, $names)) {
+                return $names[$languageCode];
+            }
+        }
```

Comment:
This definitely cannot be implemented like that, because it also duplicates Repository's internal logic. ContentType needs to be loaded with proper language code and everything should be fine.

So please check if that already happens (since I see in the diffs here that you're actually injecting proper language code) but if not, please trace back where the loading of `FieldDefinition`s occur and inject proper language there.

Keep in mind that this needs to work with 3rd party Field Types.